test/suite/test_checkpoint05.py

```diff
@@ -0,0 +1,99 @@
+# distribute this software, either in source code form or as a compiled
+# binary, for any purpose, commercial or non-commercial, and by any
+# means.
+#
+# In jurisdictions that recognize copyright laws, the author or authors
+# of this software dedicate any and all copyright interest in the
+# software to the public domain. We make this dedication for the benefit
+# of the public at large and to the detriment of our heirs and
+# successors. We intend this dedication to be an overt act of
+# relinquishment in perpetuity of all present and future rights to this
+# software under copyright law.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+# EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+# MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+# IN NO EVENT SHALL THE AUTHORS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+# OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+# ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+# OTHER DEALINGS IN THE SOFTWARE.
+#
+# test_checkpoint05.py
+# Verify that we don't accumulate a lot of checkpoints while a backup
+# cursor is open. WiredTiger checkpoints created after the backup cursor
+# should get deleted as usual.
+
+import time
+import wiredtiger, wttest
+
+def timestamp_str(t):
+    return '%x' % t
+
+class test_checkpoint04(wttest.WiredTigerTestCase):
+    conn_config = 'create,cache_size=100MB,log=(archive=false,enabled=true,file_max=100K)'
+
+    def make_key(self, n):
+        return "Key:" + str(n)
+
+    def make_value(self, n):
+        return "Value:" + str(n)
+
+    def do_updates(self, n):
+        nops = 500
+        ckpt_freq = 10
+
+        cursor = self.session.open_cursor(self.uri, None)
+        for i in range(n * nops, (n + 1) * nops - 1):
+            cursor[self.make_key(i)] = self.make_value(i)
+            if i % ckpt_freq == 0:
+                self.session.checkpoint(None)
+        cursor.close()
+
+    def count_checkpoints(self):
+        metadata_cursor = self.session.open_cursor('metadata:', None, None)
+
+        nckpt = 0
+        while metadata_cursor.next() == 0:
+            key = metadata_cursor.get_key()
+            value = metadata_cursor[key]
+            nckpt = nckpt + value.count("WiredTigerCheckpoint")
+        metadata_cursor.close()
+        return nckpt
+
+    def test_checkpoints_during_backup(self):
+        self.uri = 'table:ckpt05'
+        self.session.create(self.uri, 'key_format=S,value_format=S')
+
+        # Insert some data and take checkpoints
+        self.do_updates(0)
+
+        backup_cursor = self.session.open_cursor('backup:', None, None)
+        initial_count = self.count_checkpoints()
+
+        # Checkpoints created immediately after a backup cursor may get pinned.
+        # Pause to avoid this.
+        time.sleep(2)
+
+        # Insert some more data, then update existing values
+        self.do_updates(1)
+        self.do_updates(0)
```

Comment:
Here's where you could just loop forcing some checkpoints. No data mods needed even.

Reply:
Good to know. I was thinking we might elide a checkpoint if nothing had changed since the previous one.

Reply:
So it turns out that just calling `session.checkpoint()` in a loop with no other activity doesn't create a lot of checkpoints. So I include one insert per checkpoint now.

Reply:
Sorry, the keyword in my comment was "forcing". I thought somewhere I had typed in that, but maybe I erased it. I meant `session.checkpoint("force=true")`. Calling that in a loop should create a checkpoint each time.